operator/controllers/btpoperator_controller_test.go

```diff
@@ -367,8 +376,144 @@ var _ = Describe("BTP Operator controller", Ordered, func() {
 			doChecks()
 		})
 	})
+
+	Describe("Update", func() {
+		updateCtx := context.Background()
+
+		var initChartVersion string
+		var expectedElementsCount int
+
+		BeforeAll(func() {
+			createPrereqs()
+
+			secret, err := createCorrectSecretFromYaml()
+			Expect(err).To(BeNil())
+			Eventually(k8sClient.Create(updateCtx, secret)).WithTimeout(k8sOpsTimeout).WithPolling(k8sOpsPollingInterval).Should(Succeed())
+
+			cr = createBtpOperator()
+			Expect(k8sClient.Create(updateCtx, cr)).To(Succeed())
+			Eventually(getCurrentCrState).WithTimeout(crStateChangeTimeout).WithPolling(crStatePollingInterval).Should(Equal(types.StateProcessing))
+			Eventually(getCurrentCrState).WithTimeout(crStateChangeTimeout).WithPolling(crStatePollingInterval).Should(Equal(types.StateReady))
+
+			initChartVersion, err = ymlutils.ExtractValueFromLine(fmt.Sprintf("%s/Chart.yaml", ChartPath), "version")
+			Expect(err).To(BeNil())
+
+			gvks, err := ymlutils.GatherChartGvks(defaultChartPath)
+			Expect(err).To(BeNil())
+			expectedElementsCount = len(gvks)
+		})
+
+		BeforeEach(func() {
+			err := cp.Copy(ChartPath, updatePath)
+			Expect(err).To(BeNil())
+			ChartPath = updatePath
+			simulateRestart(updateCtx, cr)
+		})
+
+		When("update of all resources names and bump chart version", func() {
+			It("new resources (with new name) should be created and old ones removed", func() {
+				err := ymlutils.TransformCharts(updatePath, suffix)
+				Expect(err).To(BeNil())
+
+				err = ymlutils.UpdateVersion(updatePath, newChartVersion)
+				Expect(err).To(BeNil())
+
+				Expect(reconciler.chartDetails.oldChartVersion).To(Equal(initChartVersion))
+				Expect(reconciler.chartDetails.currentChartVersion).To(Equal(initChartVersion))
+
+				simulateRestart(updateCtx, cr)
+				Expect(reconciler.chartDetails.oldChartVersion).To(Equal(initChartVersion))
+				Expect(reconciler.chartDetails.currentChartVersion).To(Equal(newChartVersion))
+
+				oldCount, newCount := countResources()
+				Expect(oldCount).To(BeZero())
+				Expect(newCount >= expectedElementsCount).To(BeTrue())
+			})
+		})
+
+		When("update of all resources names and leave same chart version", func() {
+			It("new resources (with new name) should be created and old ones should stay", func() {
```

Comment:
After our discussion, rename that IT. Cause version has not changed and no update was executed then. Only old resource will stay.